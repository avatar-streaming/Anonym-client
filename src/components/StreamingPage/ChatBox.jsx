import React from "react";
import useChatting from "../../hooks/useChatting";

function ChatBox() {
  const {
    inputValue,
    updateInputValue,
    handleSubmit,
    chatList,
    chatBoxRef,
  } = useChatting();

  return (
    <div className="chat-container">
      <div ref={chatBoxRef} className="chat-box">
        {chatList.map((chat, index) => (
          <div key={index} className="chat-item">
            <span className="chat-userName">{chat.userName}</span>
            <span>: {chat.content}</span>
          </div>
        ))}
      </div>
      <div className="chat-form">
        <form className="chatting-form" onSubmit={handleSubmit}>
          <input
            type="text"
            className="input-text"
            value={inputValue}
            onChange={(e) => {
              updateInputValue(e.target.value);
            }}
          />
          <button>Chat</button>
        </form>
      </div>
    </div>
  );
}

export default ChatBox;
