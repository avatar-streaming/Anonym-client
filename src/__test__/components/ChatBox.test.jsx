import { Provider } from "react-redux";
import thunk from "redux-thunk";
import ChatBox from "../../components/StreamingPage/ChatBox";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";

describe("<ChatBox />", () => {
  const mockStore = configureMockStore([thunk]);
  let store = null;
  let component = null;

  beforeEach(() => {
    store = mockStore({
      user: {
        userInfo: {
          userName: "mock name",
        },
      },
    });
    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <ChatBox />
      </Provider>
    );
  });

  it("should render with given state from redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should chatting", () => {
    // const chattingInput = component.root.findByProps({ type: "text" });

    // renderer.act(() => {
    //   chattingInput.props.onChange({
    //     target: {
    //       value: "chatting",
    //     },
    //   });
    // });
    // expect(component.toJSON()).toMatchSnapshot();

    // renderer.act(() => {
    //   userThumnailButton.props.onClick({
    //     preventDefault: jest.fn(),
    //     isUpdateUserThumnail: store.dispatch(),
    //   });
    // });
    // expect(store.dispatch).toHaveBeenCalledTimes(1);
  });
});
