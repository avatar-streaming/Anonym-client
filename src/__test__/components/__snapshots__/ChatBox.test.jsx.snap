// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatBox /> should render with given state from redux store 1`] = `
<div
  className="chat-container"
>
  <div
    className="chat-list"
    data-testid="chat-container"
  />
  <div
    className="chat-form"
  >
    <form
      className="chat-form__form"
      onSubmit={[Function]}
    >
      <input
        className="chat-form__input input-text"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button>
        Chat
      </button>
    </form>
  </div>
</div>
`;
